main.go

```diff
@@ -6,225 +6,166 @@ import (
+	//parse array of input objects
+	input, err := parseJsonInput(listOfImage, appName)
 	if err != nil {
-		return nil, errors.Wrap(err, "error in retrieving list of deployments")
-	}
-	return spec, nil
-
-}
-
-//isDeployed checks for the status of deployment
-func (d *DoctlServices) isDeployed(appID string) error {
-	done := false
-	for !done {
-		fmt.Println("App Platform is Building ....")
-		spec, err := d.getCurrentDeployment(appID)
-		if err != nil {
-			return errors.Wrap(err, "error in retrieving list of deployments")
-		}
-		var app []godo.Deployment
-		err = json.Unmarshal(spec, &app)
-		if err != nil {
-			return errors.Wrap(err, "error in parsing deployment")
-		}
-		if app[0].Phase == "ACTIVE" {
-			fmt.Println("Build successful")
-			return nil
-		}
-		if app[0].Phase == "Failed" {
-			fmt.Println("Build unsuccessful")
-			return errors.Wrap(err, "build unsuccessful")
-		}
+		return err
 	}
-	return nil
-}
 
-//retrieveActiveDeployment retrieves currently deployed app spec of on App Platform app
-func (d *DoctlServices) retrieveActiveDeployment(appID string) ([]byte, error) {
-	cmd := exec.Command("sh", "-c", fmt.Sprintf("doctl apps get --format ActiveDeployment.ID --no-header %s", appID))
-	deployID, err := cmd.Output()
+	//parse array of deployment objects
+	appSpec, err := parseDeploymentSpec(apps)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to retrieve active deployment")
+		return err
 	}
-	deploymentID := strings.TrimSpace(string(deployID))
 
-	//get app based on appID
-	cmd = exec.Command("sh", "-c", fmt.Sprintf("doctl app get-deployment %s %s -ojson", appID, string(deploymentID)))
-	apps, err := cmd.Output()
-	if err != nil {
-		return nil, errors.Wrap(err, "error in retrieving currently deployed app id")
+	//updates all the docr images based on user input
+	newErr := filterApps(input, *appSpec)
+	if newErr.name != "" {
+		log.Print(newErr.name)
+		if len(newErr.notFound) != 0 {
+			log.Fatalf("%v", newErr.notFound)
+		}
+		return errors.New(newErr.name)
 	}
-	return apps, nil
-}
 
-//updateAppPlatformAppSpec updates app spec and creates deployment for the app
-func (d *DoctlServices) updateAppPlatformAppSpec(appID string) error {
-	cmd := exec.Command("sh", "-c", fmt.Sprintf("doctl app update %s --spec .do._app.yaml", appID))
-	_, err := cmd.Output()
-	if err != nil {
-		fmt.Print(err)
-		return errors.Wrap(err, "unable to update app")
-	}
-	err = d.createDeployments(appID)
+	//build yaml from the input json data
+	newYaml, err := yaml.Marshal(appSpec)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "Error in building yaml")
 	}
-	return nil
-}
 
-//createDeployments creates deployment for the app
-func (d *DoctlServices) createDeployments(appID string) error {
-	cmd := exec.Command("sh", "-c", fmt.Sprintf("doctl app create-deployment %s", appID))
-	_, err := cmd.Output()
+	//write to local temp file
+	err = ioutil.WriteFile(".do._app.yaml", newYaml, 0644)
```

Comment:
Since we need to write the app spec to disk, can we write it to the GH Action temp dir? That way it's cleaned up between steps.

Reply:
used https://pkg.go.dev/io/ioutil#TempFile

Reply:
I was wondering if there is any GH action specific temp dir? Right now I updated tmp file creation using https://pkg.go.dev/io/ioutil#TempFile which will create unique tmp file for our use case.